spec/controler_spec.rb

```diff
@@ -1,47 +1,50 @@
 # frozen_string_literal: true
 
 require 'controler'
-require 'user_interface'
 require 'null_action'
+require 'user_interface'
 
 RSpec.describe Controler do
-  let(:actions) { [NullAction.new] }
+  let(:actions) { [NullAction.new, NullAction.new] }
+  let(:exit_input) { StringIO.new('2') }
   let(:output) { StringIO.new }
-  let(:valid_input) { StringIO.new('1') }
 
   it 'tells the user interface to print the menu' do
-    ui = UserInterface.new(valid_input, output)
-    controler = Controler.new(ui, actions)
+    ui = UserInterface.new(exit_input, output)
+    controler = described_class.new(ui, actions)
 
     controler.start
 
     expect(output.string).to include(UserInterface::MENU_MESSAGE)
   end
 
   it 'receives the input from the user interface' do
-    ui_double = double('UserInterface', menu_choice: 1)
-    controler = Controler.new(ui_double, actions)
+    ui_double = double('UserInterface', menu_choice: 2)
+    controler = described_class.new(ui_double, actions)
 
     controler.start
 
     expect(ui_double).to have_received(:menu_choice).once
   end
 
-  it 'selects correct action to perform according to input' do
-    ui = UserInterface.new(valid_input, output)
-    controler = Controler.new(ui, actions)
+  it 'runs the actions chosen' do
+    null_action = double('NullAction', run: nil)
+    ui = UserInterface.new(StringIO.new("1\n2"), output)
+    controler = described_class.new(ui, [null_action, null_action])
 
-    expect(controler.start).to eq(nil)
-  end
+    controler.start
 
-  it 'runs the action' do
-    action_double = double('NullAction', run: nil)
+    expect(null_action).to have_received(:run).twice
+  end
 
-    ui = UserInterface.new(valid_input, output)
-    controler = Controler.new(ui, [action_double])
+  it 'prints menu after action is run' do
```

Comment:
I think I would just update the first test with the contents of this test rather than adding another one
since they are testing the same (that the menu is printed)

Also, you can use the actual actions here, you don't need the doubles :+1: